pkg/controllers/thanos-querier/controller.go

```diff
@@ -0,0 +1,353 @@
+/*
+Copyright 2021.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package thanos_querier
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	msoapi "github.com/rhobs/monitoring-stack-operator/pkg/apis/v1alpha1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/go-logr/logr"
+)
+
+type reconciler struct {
+	client.Client
+	scheme *runtime.Scheme
+	logger logr.Logger
+}
+
+// RBAC for watching monitoring stacks
+//+kubebuilder:rbac:groups=monitoring.rhobs,resources=monitoringstacks,verbs=get;list;watch
```

Comment:
We do Get MonitoringStacks when the associated and watched Service changes.
https://github.com/rhobs/monitoring-stack-operator/blob/52d403a6b08ef659448805fe60ce65907d880931/pkg/controllers/thanos-querier/controller.go#L222

Reply:
But we are getting it through the informer cache, aren't we? hence shouldn't list,watch suffice?

Reply:
On a different thought, we shouldn't need this here since the monitoring-stack controller should have specified this anyway - right :-) ?

Reply:
> But we are getting it through the informer cache, aren't we? hence shouldn't list,watch suffice?

The cache should be transparent for users, so I'd rather not rely on that.


> On a different thought, we shouldn't need this here since the monitoring-stack controller should have specified this anyway - right :-) ?

I suppose so yes and the generated rbac specification will only have it once. Having `get` in this annotation though makes the dependency explicit and independent from what other controllers specify.